cukinia

```diff
@@ -363,16 +363,45 @@ _cukinia_process()
 {
 	local pname="$1"
 	local puser="$2"
-	local ret
+	local proc_name
+	local proc_uid
 
 	_cukinia_prepare "Checking process \"$pname\" as ${puser:-any user}"
-	ret=$(ps -a -o user,comm |
-			awk '($2 == "'$pname'") && ($1 ~ /^'${puser:-.*}'$/) {
-			    print;
-			}')
-	[ -n "$ret" ] && return 0 || return 1
-}
 
+	if [ ! -z $puser ]; then
+		puser_id=`id -u $puser &>/dev/null`
+		# check if owner exist, return 1 then
+		if [ $? -ne 0 ]; then
+			return 1
+		fi
+	fi
+
+	# loop in /proc to get process name and optionally process owner
+	for p in `ls /proc/ | grep -Eo ''[[:digit:]]*''`; do
+		# ensure that a status file does exist
+		if [ -f /proc/$p/status ]; then
+			proc_name=`grep Name /proc/$p/status | cut -f 2`
+
+			# check by process name
+			if [ "$proc_name" = "$pname" ]; then
+				if [ ! -z "$puser_id" ]; then
```

Comment:
Same comment...

``` bash
for status in /proc/[0-9]*/status; do
    [ "$(grep -q Name "$status")" = "$proc_name" ] || continue

    if [ -z "$puser" ]; then
        return 0
    fi

    [ "$(grep -q Uid "$status")" = "$puser_uid" ] || continue

    return 0
done
```

Reply:
same here."cut -f 2" should be append to the grep of Uid